src/modules/openfx/mlt_openfx.c

```diff
@@ -0,0 +1,2020 @@
+  propSetInt ((OfxPropertySetHandle) clip_prop, kOfxImagePropRegionOfDefinition, 3, height);
+
+  propSetDouble ((OfxPropertySetHandle) clip_prop, kOfxImagePropPixelAspectRatio, 0, (double) width / (double) height);
+
+  propSetString ((OfxPropertySetHandle) clip_prop, kOfxImageEffectPropPreMultiplication, 0, kOfxImageUnPreMultiplied);
+  propSetString ((OfxPropertySetHandle) clip_prop, kOfxImagePropField, 0, kOfxImageFieldNone); /* I'm not sure about this */
+
+  char *tstr = calloc(1, strlen ("Source") + 11);
+  sprintf (tstr, "%s%04d%04d", "Source", rand () % 9999, rand () % 9999); /* WIP: do something better */
+  propSetString ((OfxPropertySetHandle) clip_prop, kOfxImagePropUniqueIdentifier, 0, tstr);
+  free (tstr);
+
+  propSetDouble ((OfxPropertySetHandle) clip_prop, kOfxImageEffectPropRenderScale, 0, 1.0);
+  propSetDouble ((OfxPropertySetHandle) clip_prop, kOfxImageEffectPropRenderScale, 1, 1.0);
+
+  propSetInt ((OfxPropertySetHandle) clip_prop, kOfxImageClipPropConnected, 0, 1);
+}
+
+void
+mltofx_get_clip_preferences (OfxPlugin *plugin, mlt_properties image_effect)
+{
+  mlt_properties get_clippref_args  = mlt_properties_get_data (image_effect, "get_clippref_args",  NULL);
+  OfxStatus status_code =
+     plugin->mainEntry (kOfxImageEffectActionGetClipPreferences,
+     		     (OfxImageEffectHandle) image_effect,
+     		     NULL,
+     		     (OfxPropertySetHandle) get_clippref_args);
+  mltofx_log_status_code (status_code, kOfxImageEffectActionGetClipPreferences);
+}
+
+void
+mltofx_get_regions_of_interest (OfxPlugin *plugin, mlt_properties image_effect, double width, double height)
+{
+  mlt_properties get_roi_in_args  = mlt_properties_get_data (image_effect, "get_roi_in_args",  NULL);
+  mlt_properties get_roi_out_args = mlt_properties_get_data (image_effect, "get_roi_out_args", NULL);
+
+  propSetDouble ((OfxPropertySetHandle) get_roi_in_args, kOfxPropTime, 0, 0.0);
+  propSetDouble ((OfxPropertySetHandle) get_roi_in_args, kOfxPropTime, 1, 0.0);
+  propSetDouble ((OfxPropertySetHandle) get_roi_in_args, kOfxImageEffectPropRenderScale, 0, 1.0);
+  propSetDouble ((OfxPropertySetHandle) get_roi_in_args, kOfxImageEffectPropRenderScale, 1, 1.0);
+
+  propSetDouble ((OfxPropertySetHandle) get_roi_in_args, kOfxImageEffectPropRegionOfInterest, 0, 0.0);
+  propSetDouble ((OfxPropertySetHandle) get_roi_in_args, kOfxImageEffectPropRegionOfInterest, 1, 0.0);
+  propSetDouble ((OfxPropertySetHandle) get_roi_in_args, kOfxImageEffectPropRegionOfInterest, 2, width);
+  propSetDouble ((OfxPropertySetHandle) get_roi_in_args, kOfxImageEffectPropRegionOfInterest, 3, height);
+
+  OfxStatus status_code =
+  plugin->mainEntry (kOfxImageEffectActionGetRegionsOfInterest,
+     		     (OfxImageEffectHandle) image_effect,
+     		     (OfxPropertySetHandle) get_roi_in_args,
+     		     (OfxPropertySetHandle) get_roi_out_args);
+  mltofx_log_status_code (status_code, kOfxImageEffectActionGetRegionsOfInterest);
+}
+
```

Comment:
Nit, this should come after begin sequence render per the header declaration and what end by itself means.